lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -346,6 +346,7 @@ def _on_scrape_targets_changed(self, event):
     "label_limit",
     "label_name_length_limit",
     "label_value_lenght_limit",
+    "honor_labels",
```

Comment:
@balbirthomas, we had agreed NOT to allow Providers to set this for now.

Reply:
E.g., here: https://github.com/canonical/prometheus-k8s-operator/blob/f35ff29e037dddfe826b1901d81ecb6248c1a562/src/charm.py#L335

Reply:
I find that the proposed solution in the PR violates big times the element of least surprise: we allow to set a `honor_labels` key in the relation, then we override whatever value comes in in the `MetricsEndpointConsumer`. No. Much better to hard-code it in the `charm.py` of Prometheus and NOT allow the `honor_labels` key in the relation. That way, providers are not tricked into thinking they can control that.

Reply:
Ok. Let me give this another thought. I agree with the issue of "least surprise". I put it in allowed labels because we use it to sanitize jobs. But I can ensure the label is injected after job sanitization. I think it already is but let me double check.

Reply:
Removed honor_label from allowed keys. Had to modify one test which was checking that the keys returned by `jobs()` method were a strict subset of `ALLOWED_KEYS`. Please re-review.

Reply:
Does that really address Michele's point about `prometheus-scrape-config` and such?

Reply:
Sorry missed that. Will look again.

Reply:
The honor labels is not set by the charm itself. Also added unit test to check it is always set.